runtime/src/pcr.rs

```diff
@@ -0,0 +1,92 @@
+// Licensed under the Apache-2.0 license
+
+use crate::Drivers;
+use caliptra_common::mailbox_api::{
+    IncrementPcrResetCounterReq, MailboxResp, MailboxRespHeader, QuotePcrsReq, QuotePcrsResp,
+};
+use caliptra_drivers::{
+    hand_off::DataStore, CaliptraError, CaliptraResult, Ecc384PrivKeyIn, KeyReadArgs, PcrBank,
+    PcrId,
+};
+use zerocopy::{transmute, FromBytes};
+
+pub struct PcrResetCounter {
+    counter: [u32; 32],
+}
+
+impl Default for PcrResetCounter {
+    fn default() -> Self {
+        PcrResetCounter::new()
+    }
+}
+
+impl PcrResetCounter {
+    fn new() -> PcrResetCounter {
+        PcrResetCounter { counter: [0; 32] }
+    }
+
+    pub fn get(&self, id: PcrId) -> u32 {
+        self.counter[usize::from(id)]
+    }
+
+    pub fn increment(&mut self, id: PcrId) {
+        self.counter[usize::from(id)] += 1;
```

Comment:
This can overflow and panic. Can we add a check for this? Can just fail the command if an increment will overflow.